python/cudf/cudf/core/column/column.py

```diff
@@ -192,6 +192,122 @@ def size(self) -> int:
         return self._buffer.size
 
 
+def _dtype_to_metadata(dtype: DtypeObj) -> plc.interop.ColumnMetadata:
```

Comment:
IMO this is something I'd put in `cudf/core/dtypes.py` and to import here (and then you can avoid accessing `StructDtype` and `ListDtype` through the cudf namespace)